### PR TITLE
fix windows clang compilation for ffi

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **libffi/3.4.2@**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

closes: #9954

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]
I hope it doesn't break MinGW, Intel, whatever else may run on Windows

**Comment by AndreyMlashkin** [comment]
> I hope it doesn't break MinGW, Intel, whatever else may run on Windows

adding m64 was ignored for clang and it will be also ignored for mingw :)

**Comment by AndreyMlashkin** [comment]
How to

**Comment by AndreyMlashkin** [comment]
How to restart the build?

**Comment by uilianries** [comment]
> How to restart the build?

You just did it. close/open the PR works too.

File: recipes/libffi/all/conanfile.py [hunk 1]
```diff
@@ -87,13 +87,16 @@ def _patch_sources(self):
     @contextlib.contextmanager
     def _build_context(self):
         extra_env_vars = {}
-        if self._is_msvc:
+        if tools.os_info.is_windows:
```
Comment (SpaceIm) [inline recipes/libffi/all/conanfile.py:90]: What? It will likely break MinGW.

**Comment by SpaceIm** [comment]
this approach seems neater: https://github.com/conan-io/conan-center-index/pull/9993

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**